src/Datasource/Dummy.php

```diff
@@ -0,0 +1,164 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         2.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ElasticSearch\Datasource;
+
+use Cake\Database\Driver;
+
+/**
+ * Represents a dummy driver used only for convertions
+ * between types of ElasticSearch and PHP.
+ */
+class Dummy extends Driver
```

Comment:
Why did you take this approach? The elastic search plugin doesn't use database drivers. Instead could we implement the DriverInterface?

Reply:
I tried to change as litle as possible the code base, but `Cake\Database\Type` system always expect a `Driver` instance on its methods.

As implementing a full `Driver` and separate `Connection` from him, I will need to rewrite many parts of current codebase.

Reply:
Ah. We might need a driver instance in 3.x compatible code but the 4.x compatible code should be able to use an interface.

Reply:
We should mark this `@internal` at least and probably name it `DummyDriver`.

Reply:
Current drivers doesn't include suffix _Driver_. Maybe follow namespace pattern?

`ElasticSearch\Database\Driver\Dummy.php` ?

Reply:
Moved to `Database\Driver` namespace and put a `@internal` mark.

Also, removed `strict` as 2.x branch don't have it in any file.